rust/src/coordinator.rs

```diff
@@ -255,83 +231,49 @@ pub trait Coordinators: Sized {
     /// Validate an update signature and its implied task.
     fn validate_update_task(
         &self,
+        pk: &UpdateParticipantPublicKey,
         sum_signature: &ParticipantTaskSignature,
         update_signature: &ParticipantTaskSignature,
-        pk: &UpdateParticipantPublicKey,
     ) -> Result<(), PetError> {
-        if pk.verify_detached(sum_signature, &[self.seed().as_slice(), b"sum"].concat())
+        if pk.verify_detached(sum_signature, &[self.seed.as_slice(), b"sum"].concat())
             && pk.verify_detached(
                 update_signature,
-                &[self.seed().as_slice(), b"update"].concat(),
+                &[self.seed.as_slice(), b"update"].concat(),
             )
-            && !sum_signature.is_eligible(*self.sum())
-            && update_signature.is_eligible(*self.update())
+            && !is_eligible(sum_signature, self.sum)
+            && is_eligible(update_signature, self.update)
         {
             Ok(())
         } else {
             Err(PetError::InvalidMessage)
         }
     }
 
-    /// Add a sum participant to the sum dictionary. Fails if it is a repetition.
-    fn add_sum_participant(
-        &mut self,
-        pk: &SumParticipantPublicKey,
-        ephm_pk: &SumParticipantEphemeralPublicKey,
-    ) -> Result<(), PetError> {
-        if !self.sum_dict().contains_key(pk) {
-            self.sum_dict_mut().insert(*pk, *ephm_pk);
-            Ok(())
-        } else {
-            Err(PetError::InvalidMessage)
-        }
-    }
-
     /// Freeze the sum dictionary.
     fn freeze_sum_dict(&mut self) {
-        *self.seed_dict_mut() = self
-            .sum_dict()
+        self.seed_dict = self
+            .sum_dict
             .keys()
             .map(|pk| (*pk, LocalSeedDict::new()))
             .collect();
     }
 
-    /// Aggregate a local masked model to the global masked model. Fails if the model types don't
-    /// conform.
-    fn aggregate_masked_model(&mut self, local_masked_model: &MaskedModel) -> Result<(), PetError> {
```

Comment:
the aggregation logic seems to be gone now, i didn't find where it was reintroduced, did this happen by accident? this should be reintroduced with the new tools available from the `mask` module.